File: app-ios/TutanotaSharedFramework/Utils/CancelledError.swift
public struct CancelledError: TutanotaError {
	public static let name: String = "de.tutao.tutashared.CancelledError"
	public let message: String
	public let underlyingError: (any Error)?
	public init(message: String, underlyingError: (any Error)?) {
		self.message = message
		self.underlyingError = underlyingError
	}
	public init(message: String) { self.init(message: message, underlyingError: nil) }
}

File: src/platform-kit/app-env/CancelledError.ts
import { TsString, TSwUncheckedSendable, TutanotaError } from "@tutao/lang-api"

@TSwUncheckedSendable({ reasoning: "TutanotaError is TSUncheckedSendable and CancelledError does not introduce any new fields" })
export class CancelledError extends TutanotaError {
	/**
	 * A cancelled error is thrown when a async action is aborted
	 * @param message An information about the exception.
	 * @param reason A cancellation reason.
	 */
	constructor(
		message: TsString,
		readonly reason: TsString = "unknown",
	) {
		super("CancelledError", message)
	}
}
